packages/tools-iottwinmaker/README.md

```diff
@@ -74,13 +91,23 @@ For example:
 tmdt deploy --region us-east-1 --workspace-id SyncB --dir /tmp/testproj
 ```
 
+**Note:** --workspace-id can either be an existing TwinMaker workspace, or you may enter a new ID for TMDT to automatically create your workspace (you will be prompted for confirmation). Take notice that the s3 bucket, IAM role, and dashboard role and policy is the default TwinMaker configurations generated as in the console. You may need to edit the created role/policy based on your specific use cases.
+
 ___
 
-## Nuke a workspace
+## TMDT Nuke
 
-The following will delete all entities, component types, and scenes from a workspace so that the workspace can be deleted.
+### Nuke Parameters (required)
+- *--region*: Specify the AWS region of the workspace to delete
+- *--workspace-id*: Specify the ID of the Workspace to delete from
 
-Warning: This command is destructive. Please take caution before running this command.
+### Nuke Parameters (optional flags)
+- *--delete-workspace*: Specify if TM workspace should also be deleted
+- *--delete-s3-bucket*: Specify if workspace s3 Bucket, its contents, and any associated logging bucket should be deleted
+
+The following will delete all entities, component types, and scenes from a workspace so that the workspace can be deleted. By default (no optional flags below selected), this will not delete workspace s3 bucket, IAM role, nor the TwinMaker workspace itself.
+
+**Warning:** This command is destructive. Please take caution before running this command.
```

Comment:
Is it possible to make it default to a `dry-run-mode`? i.e. they can run the script and it will give the prompts etc. but when they select "Y" it will only print out what would be deleted without actually performing the deletion

Only if they run with `--non-dry-run-mode` will it behave as it is currently

Reply:
Yes we can do this, although the easiest way I think to do this is passing in a "dry-run" boolean as an extra parameter into every single delete helper function, and reworking all the functions to call get instead of delete if that boolean is true.

Also, do you mean dry run is enabled by default for EVERY combination of nuke (regardless of what delete-workspace or delete-s3-bucket flags are there), so that it warns about entities, component types, scenes? We actually don't display which entities get deleted in "non dry run" but rather just the parent and print "waiting for recursion" so may need to consider how dry run works on entities?

Also, this change might take up to a dev day to implement and test properly, how high priority is this, compared to instead listing the s3 bucket contents before deletion like the above comment (and doesn't this conflict with that idea if we add both since they're both a form of letting the user know what gets deleted?)

Reply:
> do you mean dry run is enabled by default for EVERY combination of nuke

I'm not sure I understood the combinations aspect: I meant it as a global option. If `non-dry-run` isn't passed in, no delete operations occur regardless of what combination of delete operations are requested

> but rather just the parent and print "waiting for recursion" so may need to consider how dry run works on entities?

good point, I think we can maintain the existing approach in dry-run: basically just print that we would do recursive deletion on [entity name] -- could maybe also do something similar for the s3 one and call list-entities a few times to get a count (or print a cap of like 1000+ entities again)

> how high priority is this

I personally would put it fairly high since delete is a non-recoverable operation so would bias towards "safe default behavior" but am open to getting other opinions, especially if may take some time to implement